### PR TITLE
docs: Add  `processor.event` info to Logstash output

### PR DESCRIPTION
**Pull request by bmorelli25**
## What does this PR do?

This PR adds information about `processor.event` to the Logstash output documentation. Previously, APM Server examples used `%{[processor][event]}`, but didn't do a good job of explaining why it was necessary. This PR also makes a few grammatical changes and switches from the hardcoded `Logstash` to the attribute, `{ls}`.

## Why is it important?

This content is important because APM events should be stored in separate indices based on their event type -- which can be determined by looking at `processor.event`.

## Related issues

Closes https://github.com/elastic/apm-server/issues/3705.

### CODE REVIEW

File: libbeat/outputs/logstash/docs/logstash.asciidoc [hunk 1]
```diff
@@ -85,24 +87,24 @@ ifdef::apm-server[]
...
 
-You can access this metadata from within the Logstash config file to set values
-dynamically based on the contents of the metadata.
+In addition to metadata, {beatname_uc} provides the `processor.event` field, which
+can be used to separate {apm-overview-ref-v}/apm-data-model.html[event types] into different indices.
+endif::[]
 
+ifndef::apm-server[]
 For example, the following Logstash configuration file tells
 Logstash to use the index reported by {beatname_uc} for indexing events
```
Comment (dedemorton) [inline libbeat/outputs/logstash/docs/logstash.asciidoc:105]:
```suggestion
{ls} to use the index reported by {beatname_uc} for indexing events
```

File: libbeat/outputs/logstash/docs/logstash.asciidoc [hunk 2]
```diff
@@ -126,6 +128,10 @@ the Beat's version. For example:
 endif::[]
 
 ifdef::apm-server[]
+For example, the following {ls} configuration file tells
+{ls} to use the index and event types reported by {beatname_uc} for indexing events
+into Elasticsearch:
```
Comment (dedemorton) [inline libbeat/outputs/logstash/docs/logstash.asciidoc:133]:
```suggestion
into {es}:
```

File: libbeat/outputs/logstash/docs/logstash.asciidoc [hunk 3]
```diff
@@ -71,6 +70,9 @@ for more about the `@metadata` field.
 <2> The default is {beat_default_index_prefix}. To change this value, set the
 <<logstash-index,`index`>> option in the {beatname_uc} config file.
 <3> The current version of {beatname_uc}.
+
+You can access this metadata from within the Logstash config file to set values
```
Comment (dedemorton) [inline libbeat/outputs/logstash/docs/logstash.asciidoc:74]:
```suggestion
You can access this metadata from within the {ls} config file to set values
```

File: libbeat/outputs/logstash/docs/logstash.asciidoc [hunk 4]
```diff
@@ -85,24 +87,24 @@ ifdef::apm-server[]
...
 }
 ------------------------------------------------------------------------------
-<1> {beatname_uc} uses the `@metadata` field to send metadata to Logstash. See the
+<1> {beatname_uc} uses the `@metadata` field to send metadata to {ls}. See the
 {logstash-ref}/event-dependent-configuration.html#metadata[Logstash documentation]
 for more about the `@metadata` field.
 <2> The default is {beat_default_index_prefix}. To change this value, set the
 <<logstash-index,`index`>> option in the {beatname_uc} config file.
 <3> The default pipeline configuration: `apm`. Additional pipelines can be enabled
 with a {logstash-ref}/use-ingest-pipelines.html[Logstash pipeline config].
```
Comment (dedemorton) [inline libbeat/outputs/logstash/docs/logstash.asciidoc:96]:
```suggestion
with a {logstash-ref}/use-ingest-pipelines.html[{ls} pipeline config].
```

File: libbeat/outputs/logstash/docs/logstash.asciidoc [hunk 5]
```diff
@@ -277,7 +288,7 @@ NOTE: The "ttl" option is not yet supported on an async Logstash client (one wit
 
 ===== `pipelining`
 
-Configures number of batches to be sent asynchronously to logstash while waiting
+Configures the number of batches to be sent asynchronously to logstash while waiting
```
Comment (dedemorton) [inline libbeat/outputs/logstash/docs/logstash.asciidoc:291]:
```suggestion
Configures the number of batches to be sent asynchronously to {ls} while waiting
```

File: libbeat/outputs/logstash/docs/logstash.asciidoc [hunk 6]
```diff
@@ -277,7 +288,7 @@ NOTE: The "ttl" option is not yet supported on an async Logstash client (one wit
...
 for ACK from logstash. Output only becomes blocking once number of `pipelining`
```
Comment (dedemorton) [inline libbeat/outputs/logstash/docs/logstash.asciidoc:292]:
```suggestion
for ACK from {ls}. Output only becomes blocking once number of `pipelining`
```

File: libbeat/outputs/logstash/docs/logstash.asciidoc [hunk 7]
```diff
@@ -306,7 +317,7 @@ output.logstash:
 ===== `proxy_use_local_resolver`
 
 The `proxy_use_local_resolver` option determines if Logstash hostnames are
```
Comment (dedemorton) [inline libbeat/outputs/logstash/docs/logstash.asciidoc:319]:
```suggestion
The `proxy_use_local_resolver` option determines if {ls} hostnames are
```

**Comment by bmorelli25** [comment]
Thanks! I'll do another find and replace to make sure we got 'em all.

**Pull request merged**